src/sdl2/audio.rs

```diff
@@ -65,18 +65,19 @@ use rwops::RWops;
 
 use sys::audio as ll;
 
-impl AudioSubsystem {
+impl<'a> AudioSubsystem {
     /// Opens a new audio device given the desired parameters and callback.
     #[inline]
-    pub fn open_playback<CB, F>(&self, device: Option<&str>, spec: &AudioSpecDesired, get_callback: F) -> Result<AudioDevice <CB>, String>
-    where CB: AudioCallback, F: FnOnce(AudioSpec) -> CB
+    pub fn open_playback<CB, F, DEVICE>(&self, device: DEVICE, spec: &AudioSpecDesired, get_callback: F) -> Result<AudioDevice <CB>, String>
+    where CB: AudioCallback, F: FnOnce(AudioSpec) -> CB, DEVICE: Into<Option<&'a str>>,
```

Comment:
Why the need to add a lifetime `'a` here ? Wouldn't it work with simply `Into<Option<&str>>` ?